src/Components/WebAssembly/WebAssembly/src/Services/DefaultWebAssemblyJSRuntime.cs

```diff
@@ -47,8 +55,9 @@ public static void EndInvokeJS(string argsJson)
         });
     }
 
-    // Invoked via Mono's JS interop mechanism (invoke_method)
-    public static void BeginInvokeDotNet(string callId, string assemblyNameOrDotNetObjectId, string methodIdentifier, string argsJson)
+    [JSExport]
+    [SupportedOSPlatform("browser")]
+    public static void BeginInvokeDotNet(string? callId, string assemblyNameOrDotNetObjectId, string methodIdentifier, string argsJson)
```

Comment:
As a next step (could be next PR), this could be replaced by promise/Task instead of Begin/End pattern.

Reply:
Yes that would be good. We did the begin/end thing because it predated support for `Task` in the WebAssembly runtime's interop mechanism.